jscomp/others/js_json.ml

```diff
@@ -61,50 +61,66 @@ let test (type a) (x : 'a) (v : a kind) : bool =
   | Array -> Js_array.isArray x 
   | Object -> (Obj.magic x) != Js.null && Js.typeof x = "object" && not (Js_array.isArray x )
 
-let decodeString json = 
-  if Js.typeof json = "string" 
-  then Some (Obj.magic (json:t) : string)
+external unsafeParse : string -> t = "JSON.parse" [@@bs.val]
+
+let parse s =
+  let open Result in
+  try Ok (unsafeParse s) with
+  | e -> Error (Js.String.make e)
+
+external stringify: t -> string = "JSON.stringify" [@@bs.val]
+external stringifyAny : 'a -> string option = "JSON.stringify" [@@bs.val] [@@bs.return undefined_to_opt]
+(* TODO: more docs when parse error happens or stringify non-stringfy value *)
+
+module Decode = Js_json_decode
```

Comment:
Why avoid local open? For encoding of complex data structures it would be a lot more convenient and readable.